src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Float.cs

```diff
@@ -113,31 +115,65 @@ private static bool TryParseNormalAsFloatingPoint(ReadOnlySpan<byte> source, ref
         //
         private static bool TryParseAsSpecialFloatingPoint<T>(ReadOnlySpan<byte> source, T positiveInfinity, T negativeInfinity, T nan, out T value, out int bytesConsumed)
         {
-            if (source.Length >= 8 &&
-                source[0] == 'I' && source[1] == 'n' && source[2] == 'f' && source[3] == 'i' &&
-                source[4] == 'n' && source[5] == 'i' && source[6] == 't' && source[7] == 'y')
-            {
-                value = positiveInfinity;
-                bytesConsumed = 8;
-                return true;
-            }
+            int srcIndex = 0;
+            bool isNegative = false;
+            byte c = source[srcIndex];
 
-            if (source.Length >= 9 &&
-                source[0] == Utf8Constants.Minus &&
-                source[1] == 'I' && source[2] == 'n' && source[3] == 'f' && source[4] == 'i' &&
-                source[5] == 'n' && source[6] == 'i' && source[7] == 't' && source[8] == 'y')
+            switch (c)
             {
-                value = negativeInfinity;
-                bytesConsumed = 9;
-                return true;
+                case Utf8Constants.Minus:
+                {
+                    isNegative = true;
+                    goto case Utf8Constants.Plus;
+                }
+
+                case Utf8Constants.Plus:
+                {
+                    srcIndex++;
+                    break;
+                }
+
+                default:
+                {
+                    break;
+                }
```

Comment:
Why have the default:break?

Reply:
Was just copied from the handling that the regular parsing logic had.

Reply:
I'll remove.